### PR TITLE
Make dot work

### LINKED ISSUE #3: `dot` not defined
```
using LinearAlgebra
using LinearFractional
using JuMP
using Clp

lfp = LinearFractionalModel(solver=ClpSolver())
a = [-2, 0]
x = @variable(lfp, [i=1:2], basename="x", lowerbound=0)
y = dot(a, x)
```

yields

```
ERROR: MethodError: no method matching one(::Type{LinearFractional.LinearFractionalVariable})
Closest candidates are:
  one(::Type{Variable}) at /Users/ifiske/.julia/packages/JuMP/Xvn0n/src/JuMP.jl:515
  one(::BitArray{2}) at bitarray.jl:392
  one(::Missing) at missing.jl:79
  ...
Stacktrace:
 [1] _dot(::Array{Int64,1}, ::Array{LinearFractional.LinearFractionalVariable,1}) at /Users/ifiske/.julia/packages/JuMP/Xvn0n/src/operators.jl:305
 [2] dot(::Array{Int64,1}, ::Array{LinearFractional.LinearFractionalVariable,1}) at /Users/ifiske/.julia/packages/JuMP/Xvn0n/src/operators.jl:297
 [3] top-level scope at none:0
```

### PR DESCRIPTION
**Pull request by ianfiske**
Fixes #3

### CODE REVIEW

**Review by juliebuard — APPROVED** [review]
LinearAlegebra and  julia 1.0 !

File: test/arrayvars.jl [hunk 1]
```diff
@@ -87,3 +87,18 @@ end
     @test getvalue(xm) ≈ getvalue(x)
     @test getobjectivevalue(lfp) ≈ getobjectivevalue(m)/5.0
 end
+
+
+@testset "Misc array features" begin
```
Comment (doorisajar) [inline test/arrayvars.jl:92]: 👍

File: test/runtests.jl [hunk 2]
```diff
@@ -1,5 +1,6 @@
 using Test
 using LinearFractional
+using LinearAlgebra
```
Comment (kenahoo) [inline test/runtests.jl:3]: Is this change part of the same issue, or just something you noticed was needed?

File: test/arrayvars.jl [hunk 3]
```diff
@@ -87,3 +87,18 @@ end
...
+    lfp = LinearFractionalModel(solver=ClpSolver())
+    a = [-2, 0]
+    x = @variable(lfp, [i=1:2], basename="x", lowerbound=0)
+    @constraint(lfp, x[2] <= 6)
+    @constraint(lfp, -x[1] + x[2] <= 4)
+    @constraint(lfp, 2x[1] + x[2] <= 14)
+    @numerator(lfp,  :Min, dot(a, x) + 2)
+    @denominator(lfp,  x[1] + 3x[2] + 4)
+    @test solve(lfp) == :Optimal
+    @test all(getvalue(x) .≈ [7.0, 0.0])
```
Comment (kenahoo) [inline test/arrayvars.jl:103]: Love the Unicode operator.

File: test/runtests.jl (hunk 2 above)
Comment (ianfiske) [reply test/runtests.jl:3]: With Julia 1.0, we need to import `LinearAlgebra` to access linear algebra functions like `dot`.  We just didn't access any linear algebra in the tests prior to this new test.

Comment (kenahoo) [reply test/runtests.jl:3]: Oh, I see, thanks.

**Pull request merged**